sky/serve/replica_managers.py

```diff
@@ -819,6 +821,27 @@ def _handle_preemption(self, info: ReplicaInfo) -> bool:
     # ReplicaManager Daemon Threads #
     #################################
 
+    class ReplicaRecordAction(Enum):
+        KEEP = auto()
+        REMOVE_SCALE_DOWN = auto()
+        REMOVE_PREEMPTION = auto()
+        REMOVE_VERSION_MISMATCH = auto()
```

Comment:
```suggestion
    class ReplicaRecordRemoveReason(Enum):
        SCALE_DOWN = 'scale_down'
        PREEMPTION = xxx
        VERSION_MISMATCH = xxx
```
nit: how about we only keep the remove reason and return a `None` in the `get_replica_record_action` function to indicate keep it? Please also rename this function accordingly.
also, use a str for its value for more information.